### PR TITLE
Fix deprecations

### PR DESCRIPTION
**Pull request by wilzbach**
They were getting annoying ;-)

### CODE REVIEW

File: src/analysis/useless_assert.d [hunk 1]
```diff
@@ -42,11 +43,11 @@ class UselessAssertCheck : BaseAnalyzer
 		if (!skipSwitch) switch (token.type)
 		{
 		case tok!"doubleLiteral":
-			if (!token.text.removechars("Ll").to!double)
+			if (!token.text.asLowerCase.filter!(c => c != 'l').to!double)
```
Comment (ZombineDev) [inline src/analysis/useless_assert.d]: I think the following may be shorter/faster:
```D
auto filterChars(string chars, S)(S str)
{
	import std.algorithm.comparison : among;
    import std.algorithm.iteration : filter;
    import std.meta : aliasSeqOf;
	return str.filter!(c => !c.among(aliasSeqOf!chars));
}

if (!token.text.filterChars!"Ll".to!double)
    // ...

```

**Comment by wilzbach** [comment]
> I think the following may be shorter/faster:

Yes, that's very elegant!

**Comment by ZombineDev** [comment]
(Don't mind my stupid ws commits, I'll squash them out of existance as soon as the tests pass ;) )

**Comment by wilzbach** [comment]
> (Don't mind my stupid ws commits, I'll squash them out of existance as soon as the tests pass ;) )

No worries, I also made the same mistake here. Setting up a `.editorconfig` plugin in your favorite IDE can really help ;-)
Btw don't mind the test failure on dmd-nightly for now - it's a [known](https://issues.dlang.org/show_bug.cgi?id=17506) regression in EMSI containers.

**Comment by ZombineDev** [comment]
>No worries, I also made the same mistake here. Setting up a .editorconfig plugin in your favorite IDE can really help ;-)

Ah yes, I have done so ages ago, but sometimes I'm too lazy and I edit PR with the GitHub online editor.

> Btw don't mind the test failure on dmd-nightly for now - it's a known regression in EMSI containers.

I noticed, but I can't make sense of the error - I don't think it should be there. Anyway, I think we should also bring emsi_containers to the org, so we can more easily manage such issues.

**Comment by wilzbach** [comment]
> I noticed, but I can't make sense of the error - I don't think it should be there.

It was introduced here: https://github.com/dlang/dmd/pull/6816

>  Anyway, I think we should also bring emsi_containers to the org, so we can more easily manage such issues.

That's a great idea -> https://github.com/economicmodeling/containers/issues/81

**Pull request merged**